service/multisig_service.go

```diff
@@ -166,22 +169,41 @@ func (s *multisigService) CreateMultisigTx(multisigTxArgs *dto.MultisigTxArgs) (
 		Owners:            multisigTxOwners,
 		ParentTransaction: parentTransaction,
 	}
-	_, err = s.dao.CreateMultisigTx(&multisigTx)
 
+	// if tx already exists and is expired, update it
+	if tx, e := s.GetMultisigTxIgnoreState(id); e == nil {
+		log.Printf("An identical expired tx (id=%s) has been found and will be archived.\n", id)
+		_, err = s.updateExpiredMultisigTx(now, tx)
+		if err != nil {
+			return nil, err
+		}
+	}
+	_, err = s.dao.CreateMultisigTx(&multisigTx)
 	if err != nil {
 		return nil, err
 	}
 	return s.GetMultisigTx(id)
 }
 
+func (s *multisigService) updateExpiredMultisigTx(now time.Time, multisigTx *model.MultisigTx) (string, error) {
+	if multisigTx.Expiration != nil && multisigTx.Expiration.After(now) {
+		return "", ErrCannotUpdateNonExpiredTx
+	}
+	newId := multisigTx.Expiration.Format("2006-01-02T15:04:05") + "_" + multisigTx.Id
+
+	log.Printf("New id '%s' generated for expired tx with old id = %s", newId, multisigTx.Id)
+	// update multisig_tx id
+	return newId, s.dao.DeleteTxOwnersAndUpdateID(multisigTx.Id, newId)
+}
+
 func (s *multisigService) GetAllMultisigTxForAlias(alias string, timestamp string, signature string) (*[]model.MultisigTx, error) {
```

Comment:
Why do we have a new parameter (activeOnly) in internal function but are not allowed to control it via API?
IMO it should be possible to add a new (optional) field in the service method which gives us this new control.
Omitting this new field (false / undefined) should lead to the results which we got before (I think it is "true")

-> undefined: true, else (if passed) take the value and pass as activeOnly.

Reply:
The public methods of the API have remained unchanged since there is no requirement to provide such functionality to the external consumer. In my eyes and based on the fact that this ticket addresses an existing bug, the new update logic constitutes inner works of the API and thus was not made publicly accessible. I can however change that if there's a reason/use case where we would need it.